twoliter/embedded/Makefile.toml

```diff
@@ -362,7 +362,15 @@ for ws in sources variants .; do
   [ -s "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.lock" ] || continue
   CARGO_MANIFESTS+=(--sync "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml")
 done
-cargo vendor --locked --no-delete  --versioned-dirs --respect-source-config  ${CARGO_MANIFESTS[@]} .cargo/vendor > .cargo/config.toml 2> .cargo/vendor.log || {
+if [ -f ".cargo/config.toml" ]; then
+  # if the user has a configuration we should respect it
+  cp ".cargo/config.toml" ".cargo/twoliter_cargo_config.toml"
+elif [ -f ".cargo/twoliter_cargo_config.toml" ]; then
+  # If this file already exists from a previous run remove it to avoid duplication
+  # of vendor configuration
+  rm .cargo/twoliter_cargo_config.toml
+fi
+cargo vendor --locked --no-delete  --versioned-dirs --respect-source-config  ${CARGO_MANIFESTS[@]} .cargo/vendor >> .cargo/twoliter_cargo_config.toml 2> .cargo/vendor.log || {
```

Comment:
This restores functionality from the original PR that I didn't understand the motivation for. What do we mean by "we should respect it"?

In what way are we respecting it? How do we know that what we're respecting won't conflict with what we need to happen here as far as vendoring crates?

It might be better to ignore the edits made by `cargo vendor` to the config and just generate our own configuration with the expected paths whenever we want to use the vendored directory.

But then I have questions around how we can make sure the various `Makefile.toml` tasks use the vendored crates instead of downloading them all again. And how we use those crates when running cargo commands directly under `sources`.

Reply:
I mostly just add it by default, in cause an external customer wanted to use some other argument in .cargo/config.toml such as [net]git.fetch_with_cli or something